pkg/utils/tieredstore/tiered_store.go

```diff
@@ -20,26 +20,36 @@ import (
 
 	"github.com/fluid-cloudnative/fluid/pkg/common"
 	"github.com/fluid-cloudnative/fluid/pkg/ddc/base"
-	"github.com/go-logr/logr"
 	"k8s.io/apimachinery/pkg/api/resource"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
-var log logr.Logger
+var log = ctrl.Log.WithName("tieredStore")
 
-func init() {
-	log = ctrl.Log.WithName("tieredstore")
+type sortMediumType []common.MediumType
+
+func (s sortMediumType) Len() int {
+	return len(s)
 }
 
-type sortMediumType []common.MediumType
+func (s sortMediumType) Swap(i, j int) {
+	if i < s.Len() && j < s.Len() {
+		s[i], s[j] = s[j], s[i]
+	} else {
+		log.Info("index out of bound, will not swap", "i", i, "j", j)
```

Comment:
suggest:
If the index overflows, it explicitly returns an error,
This way the caller knows there is a problem with the index

Reply:
> suggest:
> If the index overflows, it explicitly returns an error,
> This way the caller knows there is a problem with the index

This function is an implement of an interface, so I cannot change the return.

Reply:
I think we don't need to check index overflow. Please see https://yourbasic.org/golang/how-to-sort-in-go/ . And its implementation in https://github.com/golang/go/blob/master/src/sort/sort.go#L231

Reply:
@yangyuliufeng Have some known error?